target_chains/sui/contracts/sources/pyth.move

```diff
@@ -85,8 +92,50 @@ module pyth::pyth {
         sources
     }
 
-    /// Create and share new price feed objects if they don't already exist.
-    public fun create_price_feeds(
+    /// Create and share new price feed objects if they don't already exist using accumulator message.
+    public fun create_price_feeds_using_accumulator(
+        pyth_state: &mut PythState,
+        accumulator_message: vector<u8>,
+        vaa: VAA, // the verified version of the vaa bytes encoded within the accumulator_message
+        clock: &Clock,
+        ctx: &mut TxContext
+    ){
+        // This capability ensures that the current build version is used.
+        let latest_only = state::assert_latest_only(pyth_state);
+
+        // Check that the VAA is from a valid data source (emitter)
+        assert!(
+            state::is_valid_data_source(
+                pyth_state,
+                data_source::new(
+                    (vaa::emitter_chain(&vaa) as u64),
+                    vaa::emitter_address(&vaa))
+                ),
+            E_INVALID_DATA_SOURCE
+        );
+
+        // decode the price info updates from the VAA payload (first check if it is an accumulator or batch price update)
+        let accumulator_message_cursor = cursor::new(accumulator_message);
+        let header = deserialize::deserialize_u32(&mut accumulator_message_cursor);
+
+        let _price_infos = vector::empty<PriceInfo>();
```

Comment:
why starting with underscore? it's an indicator of unused variable.

Reply:
because there is a compile warning:

```
warning[W09003]: unused assignment
    ┌─ ./sources/pyth.move:121:13
    │
121 │         let price_infos = vector::empty<PriceInfo>();
    │             ^^^^^^^^^^^ Unused assignment for variable 'price_infos'. Consider removing, replacing with '_', or prefixing with '_' (e.g., '_price_infos')
```

if we do

```
        let price_infos = vector::empty<PriceInfo>();
        if ((header as u64) == PYTHNET_ACCUMULATOR_UPDATE_MAGIC) {
            price_infos = accumulator::parse_and_verify_accumulator_message(&mut accumulator_message_cursor, vaa::take_payload(vaa), clock);
        }
        else {
            abort E_INVALID_ACCUMULATOR_HEADER
        };
```

Reply:
I rewrote the code like this to avoid this situation where the compiler gets tripped up by the if statement:

```
if ((header as u64) != PYTHNET_ACCUMULATOR_UPDATE_MAGIC) {
            abort E_INVALID_ACCUMULATOR_HEADER
        };
        let price_infos = accumulator::parse_and_verify_accumulator_message(&mut accumulator_message_cursor, vaa::take_payload(vaa), clock);
```